src/vs/workbench/contrib/output/common/outputChannelModel.ts

```diff
@@ -93,13 +102,65 @@ export abstract class AbstractFileOutputChannelModel extends Disposable implemen
 			this.modelUpdater.cancel();
 			this.onUpdateModelCancelled();
 		}
+		if (this.replaceAllCancellationToken) {
+			this.replaceAllCancellationToken.cancel();
+			this.replaceAllCancellationToken = null;
+			this.replaceAllValue = null;
+		}
 		if (this.model) {
 			this.model.setValue('');
 		}
 		this.endOffset = isNumber(till) ? till : this.endOffset;
 		this.startOffset = this.endOffset;
 	}
 
+	replaceAll(till: number, value: string): void {
+		if (this.modelUpdater.isScheduled()) {
+			this.modelUpdater.cancel();
+			this.onUpdateModelCancelled();
+		}
+		this.replaceAllValue = value;
+		this.replaceAllValueSeenInAppend = false;
+		this.endOffset = isNumber(till) ? till : this.endOffset;
+		this.startOffset = this.endOffset;
+
+		if (this.model) {
+			if (this.replaceAllCancellationToken) {
+				this.replaceAllCancellationToken.cancel();
```

Comment:
nit - modern TypeScript/JavaScript allows to write `this.replaceAllCancellationToken?.cancel();`

Reply:
Updated to use optional chaining operator.